### DISCUSSION TITLE
The method onUse(World, PlayerEntity, BlockHitResult) of type ExampleBlock must override or implement a supertype method

### DISCUSSION THREAD

**Discussion by rackodo** [FabricMC/fabric-api#3849] [Mod Dev Support] [answered]
### Description

Trying to follow the Fabric Mod wiki to make a basic mod and ran into an issue when copying in code.

Under the [Creating a Custom Block Class](https://fabricmc.net/wiki/tutorial:blocks#creating_a_custom_block_class) section, it provides a snippet of code that supposedly makes the block say "Hello, world!" when it's right clicked. Copying that code into my ExampleBlock.java file raises a bug.

**My code, copied from the wiki (with package imports):**
```java
package dev.rackodo.minesillies;

import net.minecraft.block.Block;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.text.Text;
import net.minecraft.util.ActionResult;
import net.minecraft.util.hit.BlockHitResult;
import net.minecraft.world.World;

public class ExampleBlock extends Block {
	public ExampleBlock(Settings settings) {
		super(settings);
	}

	@Override
	public ActionResult onUse(World world, PlayerEntity player, BlockHitResult hit) {
		if (!world.isClient) {
			player.sendMessage(Text.literal("Hello, world!"), false);
		}

		return ActionResult.SUCCESS;
	}
}
```

Error occurs on `onUse(World world, PlayerEntity player, BlockHitResult hit)` and reads:
```
The method onUse(World, PlayerEntity, BlockHitResult) of type ExampleBlock must override or implement a supertype method
```

This is a 1.20.6 mod generated from the Template Mod Generator.

**Comment by rackodo** [comment, marked as answer]
UPDATE:

Resolved by matching the onUse method with the one found in `AbstractBlock.class`.

Changed:
`onUse(World world, PlayerEntity player, BlockHitResult hit)`
to:
`onUse(BlockState state, World world, BlockPos pos, PlayerEntity player, BlockHitResult hit)`

**Comment by Daomephsta** [comment]
I'm not sure what's up with the tutorial. `onUse(BlockState state, World world, BlockPos pos, PlayerEntity player, BlockHitResult hit)` is the correct signature in 1.20.4, 1.20.5, and 1.20.6.

It's pretty normal for the tutorials to be outdated though, particularly with Mojang's current rapid release pace. This seems to be a mistake though, which should be fixed.

**Reply by rackodo** [reply]
Took the liberty of updating the wiki to match my fix.
